tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -15243,17 +15243,14 @@ void DiagnoseNodeEntry(Sema &S, FunctionDecl *FD, HLSLShaderAttr *Attr) {
     // arrays of NodeOutput or EmptyNodeOutput are not supported as node
     // parameters
     if (ParamTy->isArrayType()) {
```

Comment:
In fact, I don't think that the array variants are meant to be used as a native array type either. The way this error is written, it might sound like you just swap one type for the other. I'm not sure that will produce an error. I believe it should. I don't see any tests that try adding array braces to *Array types and my simple experiment ended with asserts.